src/Core/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -451,7 +451,8 @@ private OpenApiOperation CreateBaseOperation(string description, List<OpenApiTag
             if (dbObject.SourceType == EntitySourceType.StoredProcedure)
             {
                 Entity entityTest = _runtimeConfig.Entities[entityName];
-                List<SupportedHttpVerb> spRestMethods = entityTest.Rest.Methods.ToList();
+
+                List<SupportedHttpVerb>? spRestMethods = (entityTest.Rest.Methods is not null) ? entityTest.Rest.Methods.ToList() : null;
```

Comment:
Do we need to check for nullability of entity.rest.methods even for stored procedures? As per this code block, it doesn't look like we need to: https://github.com/Azure/data-api-builder/blob/83e300fa86f528e49a7deff09aa8605e10f39be6/src/Config/ObjectModel/RuntimeEntities.cs#L134

Reply:
Right, while it is true that Rest Methods is guaranteed to be populated with the current logic, a null check acts as an extra guard - if the deserialization logic were to change in the future in which the Rest Methods could not be populated.

IMO, adding a null check does no harm, plus brings additional defence